packages/block-library/src/image/edit.js

```diff
@@ -438,12 +455,18 @@ class ImageEdit extends Component {
 			);
 		}
 
-		const classes = classnames( className, {
+		let classes = classnames( {
 			'is-transient': isBlobURL( url ),
 			'is-resized': !! width || !! height,
 			'is-focused': isSelected,
 		} );
 
+		if ( sizeSlug ) {
+			const mediaSizeClass = 'size-' + sizeSlug;
+			const classNameWithoutSize = classes.replace( /(\s+?)size-(.*?)(\s+?)/, '' );
+			classes = classnames( classNameWithoutSize, mediaSizeClass );
+		}
```

Comment:
This will not remove other size classes. `classnames` won't replace size-thumbnail with size-large but just add size-large to the list and we don't want that.

Reply:
I'm talking about modifying the call to `classnames()` on line 458. React's reconciler will do the work of removing or adding class names as necessary.